Automattic/wp-calypso
# Plans 2023: Featured badge for current plan

Fixes Automattic/martech#1492

## Proposed Changes

Follow-up to https://github.com/Automattic/wp-calypso/pull/73436 that replaces badges with "your plan" when any one of them is the site's current plan. 

**Notes:** 

- The refactors in [plan-comparison-grid.tsx](https://github.com/Automattic/wp-calypso/pull/73582/files#diff-b28a2a8235dc8fde181278f2cb9c570052550c5276652dc66727943605f37030) may seem daunting, but they mainly moved things out of for-loops and into function components to use hooks. Some duplication around the storage feature row was also cleaned out. Not much other improvement than that (unfortunately).
- Some last-minute changes went into how the top table renders the badges, so should be tested thoroughly with different plans to make sure nothing's missed. 

**Rant:** I'll leave out the rant 💣 


## Media

<img width="700" alt="Screenshot 2023-02-23 at 3 26 03 PM" src="https://user-images.githubusercontent.com/1705499/220921251-312028dd-8536-44d1-abba-e2e56cc6a7e9.png">

<img width="700" alt="Screenshot 2023-02-23 at 4 40 55 PM" src="https://user-images.githubusercontent.com/1705499/220939135-7b5421dd-62f9-477c-bdc8-4fdc6cf01fc6.png">

<img width="500" alt="Screenshot 2023-02-23 at 3 27 19 PM" src="https://user-images.githubusercontent.com/1705499/220921537-8dc9e900-194b-470a-90b2-1e65285b98fc.png">

<img width="500" alt="Screenshot 2023-02-23 at 3 27 32 PM" src="https://user-images.githubusercontent.com/1705499/220921564-14497255-a54f-4b52-aad4-0f9ec7af3b16.png">


## Testing Instructions



- go to /plans/[SITE] with ?flags=onboarding/2023-pricing-grid and confirm the changes
- got to /start/plans and confirm the same
- try with different plans and ensure the rounded corners at the top/highlight badges render properly.

## Pre-merge Checklist
